driver/main.c

```diff
@@ -196,9 +196,12 @@ static struct tracepoint *tp_sched_switch;
 static struct tracepoint *tp_signal_deliver;
 #endif
 #ifdef CAPTURE_PAGE_FAULTS
+// Even in kernels that can support page fault tracepoints, tracepoints may be
+// disabled so check if g_fault_tracepoint_disabled is set.
 static struct tracepoint *tp_page_fault_user;
 static struct tracepoint *tp_page_fault_kernel;
 static bool g_fault_tracepoint_registered;
+static bool g_fault_tracepoint_disabled = false;
```

Comment:
Minor: initializing a static to false is not good practice according to kernel coding conventions, if you run your patch against `checkpatch.pl` in the kernel tree it should complain.